examples/factory-contract-global/src/lib.rs

```diff
@@ -0,0 +1,206 @@
+use near_sdk::PromiseError;
+use near_sdk::{env, ext_contract, near, AccountId, NearToken, Promise};
+
+#[derive(Default)]
+#[near(contract_state)]
+pub struct GlobalFactoryContract {
+    /// Store the hash of deployed global contracts for reference
+    pub deployed_global_contracts: std::collections::HashMap<String, Vec<u8>>,
+    /// Store account IDs that have deployed global contracts
+    pub global_contract_deployers: std::collections::HashMap<String, AccountId>,
+}
+
+// Example contract interface that we'll deploy as a global contract
+#[ext_contract]
+pub trait ExtStatusMessage {
+    fn set_status(&mut self, message: String);
+    fn get_status(&self, account_id: AccountId) -> Option<String>;
+}
+
+#[near]
+impl GlobalFactoryContract {
+    /// Deploy a global contract with the given bytecode, identifiable by its code hash
+    #[payable]
+    pub fn deploy_global_contract(
+        &mut self,
+        name: String,
+        code: Vec<u8>,
+        account_id: AccountId,
+        amount: NearToken,
```

Comment:
Let's not recommend users to pass the amount of tokens via arguments. Use the attached_deposit amount.